### PR TITLE
Better PG support + Faker Integration + Optional DB setup

### PR DESCRIPTION
**Pull request by rapito**
The main feature of this PR is a couple new configuration options that enable the user to scrub and load data without actually recreating the local or target DB on each run.

There's also additional features added to this described in the **Changes** section.

## Reason 

- Often we need to get DB configurations on multiple environments. As the gem is right now (master) you have to scrub the data and then load it on ALL environments each time. This prevents that by simply sharing one scrubbed file and then loading it on-demand on each env.
i.e:  Scrub from Production, Load on Staging, QA, Dev 1, Dev 2, Dev 3, etc...

- Sometimes we only need to import the data without deleting the DB or the schema.

## Changes:

- Added `recreate_db` config option which:
  - Allows to skip dumping structures and migrations on `Brillo.scrub!`
  - Allows to skip db recreation on `Brillo.load!`
- Added `filename` scrub/load option to allow changing the name of the resulting scrubbed file or file to import on the DB.
- Added config option `schema_search_path` support for Postgres so you can specify the schema (if any) on your current pg db.
- Added multiple Faker scrubber for general purpose usage.
- Added `keep_local` param to `Brillo.load` to prevent downloading a file over and over again if the filename provided is already present. (useful for test runners)

### CODE REVIEW

**Review by bessey — CHANGES_REQUESTED** [review]
Hey, first off, thank you for the contribution! To tell the truth I don't use this library anymore but nevertheless I am happy to review and merge the majority of whats in this PR. I have a couple questions inline, and the Faker stiff specifically I do not wish to take on as a dependency, but the rest broadly looks good to me.

File: lib/brillo/scrubber.rb
Comment (rapito) [reply lib/brillo/scrubber.rb:30]: Yeah definitely, it makes a lot of sense.

File: README.md
Comment (rapito) [reply README.md:46]: > Is this not the default behaviour? I don't see a need to put it in the example if it is. Convention over configuration and all that

 I added it to be able to document it as it is kind of hidden away. I found it while I was building the other config variables and noticed it was useful and I really didn't knew about it since It's not mentioned anywhere else on the readme.

File: lib/brillo.rb
Comment (rapito) [reply lib/brillo.rb:49]: Of course!

File: lib/brillo/adapter/postgres.rb
Comment (rapito) [reply lib/brillo/adapter/postgres.rb:7]: Not really, it was just the first instance of configuration I found to specify the schema search path on `psql`. 
I think I tried with other cli options but they didn't work all the time so I just left this one. I don't really mind along as it works consistently lol.

Comment (rapito) [reply lib/brillo/adapter/postgres.rb:7]: Not really, it was just the first instance of configuration I found to specify the schema search path on psql.
I think I tried with other cli options but they didn't work all the time so I just left this one. I don't really mind along as it works consistently lol.

**Comment by rapito** [comment]
Applied the requested changes, hopefully it's good now :D

**Comment by bessey** [comment]
Hey, I've updated the master branch to automatically run tests via Github Actions. Could you catch up (and populate the changelog with an "Unreleased" section of your changes? I will cut a version as soon as I merge. 2.1.0 I guess?

**Comment by rapito** [comment]
> Hey, I've updated the master branch to automatically run tests via Github Actions. Could you catch up (and populate the changelog with an "Unreleased" section of your changes? I will cut a version as soon as I merge. 2.1.0 I guess?

Sure no problem!

**Comment by rapito** [comment]
Added the changes to the Changelog

**Review by bessey — APPROVED** [review]
Nice, thanks for this! I will merge now and cut the gem when I'm at a my main computer

**Pull request merged**
